org-roam.el

```diff
@@ -395,6 +395,51 @@ which takes as its argument an alist of path-completions.  See
   (interactive)
   (find-file org-roam-directory))
 
+;;;; org-roam-find-index
+(defcustom org-roam-index-file nil
+  "Path to the Org-roam index file.
+The path can be a string or a function.  If it is a string, it
+should be the path (absolute or relative to `org-roam-directory')
+to the index file.  If it is is a function, the function should
+return the path to the index file.  Otherwise, the index is
+assumed to be a note in `org-roam-directory' whose title is
+'Index'."
+  :type '(choice
+          (string :tag "Path to index" "%s")
+          (function :tag "Function to generate the path"))
+  :group 'org-roam)
+
+(defun org-roam--get-index-path ()
+  "Return the path to the index in `org-roam-directory'.
+The path to the index can be defined in `org-roam-index-file'.
+Otherwise, it is assumed to be a note in `org-roam-directory'
+whose title is 'Index'."
+  (let* ((index org-roam-index-file)
+         (path (pcase index
+                 ((pred functionp) (funcall index))
+                 ((pred stringp) index)
+                 ('nil (user-error "You need to set `org-roam-index-file' before you can jump to it"))
+                 (wrong-type (signal 'wrong-type-argument
+                                     `((functionp stringp)
+                                       ,wrong-type))))))
+    (pcase index
+      ((pred (f-relative-p)) (concat (file-truename org-roam-directory) path))
+      (_ index))))
```

Comment:
```suggestion
    (if (f-relative-p index)
    (concat (file-truename org-roam-directory) path)
  index)
```
pcase is usually overkill when there's a single if/else branch.

Reply:
Thanks, I forgot to refactor it after removing the conditions that warranted it.

Fixed in f4399737271ffde04cb17179e2e3dda83be39514.